tests/flows/jobs/test_synchronize_agreements.py

```diff
@@ -665,3 +665,30 @@ def test_sync_agreement_accounts_skip_management_account(
     mock_client.list_accounts.return_value = aws_accounts_factory(accounts=accounts)
     sync_agreement_subscriptions(mpt_client, aws_client, mock_agreement, False)
     mock_client.list_accounts.assert_called_once()
+
+
+def test_synchronize_agreements_exception(
+    mocker,
+    mpt_client,
+    config,
+    agreement_factory,
+):
+    mocker.patch(
+        "swo_aws_extension.flows.jobs.synchronize_agreements.sync_agreement_subscriptions",
+        side_effect=Exception("Test exception"),
+    )
+    mock_agreement = agreement_factory(vendor_id="123456789012")
+    mocker.patch(
+        "swo_aws_extension.flows.jobs.synchronize_agreements.get_agreements_by_query",
+        return_value=[mock_agreement],
+    )
+    exception_logger = mocker.patch(
```

Comment:
I think you can check here `capsys`

https://docs.pytest.org/en/stable/reference/reference.html#std-fixture-capsys

that everything is in stderr